src/parser/parser.h

```diff
@@ -1,16 +1,25 @@
 #pragma once
 
 #include "lexer/lexer.h"
+#include "error/error.h"
 
 namespace parser {
-// Different error types that can be returned from parsing.
-enum class ParseErrorCode {
-  VALID,
-  SYNTAX_ERROR,
-};
 
-// Error produced while parsing (TODO: make it more than an enum)
-using ParseError = ParseErrorCode;
+class ParseError : public Error {
+ public:
+  ParseError(const lexer::LexError& lexerror) : Error(lexerror.toString()) {}
+
+  explicit ParseError(const std::string& message, const lexer::Token& t)
+    : Error(location_line(message, t))
+  {}
+
+  static std::string location_line(const std::string& message,
+                                   const lexer::Token& t) {
+    std::stringstream ss;
+    ss << message << "\n At " << t.begin.toString();
```

Comment:
Maybe it's better to indicates "Between {begin} and {end}". In which case, you probably want to avoid the repetition of the file name.